### DISCUSSION TITLE
The second request cannot accept the message

### DISCUSSION THREAD

**Discussion by tsiens** [hbldh/bleak#1483] [Q&A]
* bleak version:0.21.1
* Python version:3.10.0
* Operating System:python

### Description
``` python
from bleak import BleakClient
import asyncio
async def connect(addr,char):
    def notification_handler(_,r):
        print(r)
    async with BleakClient(addr,timeout=10) as client:
        await client.start_notify(char, notification_handler)
        await asyncio.sleep(3)
        await client.stop_notify(char)

while True:
    asyncio.run(connect(addr,char))
    if input(f'conitue')!='':
        break
```

When I first connect, I can receive notifications，But when I press ‘Enter‘ for the second run, I can't see any output.
I restarted the program, but there is still no message.
I switched to a visualization software and it's the same.
After restarting the Bluetooth device,  i can receive messages without exiting the program.
I think it's my first time not really ending the connection, and I've been occupying it all along.
I don't know if this is a problem with bleak or my Bluetooth device
What should I do？
tks

---------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------
**update：**
```python
from bleak import BleakClient,BleakScanner
import asyncio
async def connect(addr,char):
    def notification_handler(_,r):
        print(r)
    async with BleakClient(addr,timeout=10) as client:
        await client.start_notify(char, notification_handler)
        await asyncio.sleep(3)
        await client.stop_notify(char)
async def scan():
    for i in await BleakScanner.discover():
        print(i)

    
async def main():
    await scan()
    await connect(addr,char)

asyncio.run(main())
```
I modified the code, and on the second attempt, "scan" can run but "connect" not run

**Comment by dlech** [comment]
`asyncio.run()` should only be used once per program. There is an `aioconsole` library that you can use to get input in a non-blocking way if you need to.

**Comment by tsiens** [comment]
> `asyncio.run()` should only be used once per program. There is an `aioconsole` library that you can use to get input in a non-blocking way if you need to.

Thank you for your prompt response，That's the problem with my code.
For the aioconsole , I couldn't find sufficient documentation on Google. How should I rewrite it based on my functional requirements? Thank you
